pkg/rpcclient/rpc.go

```diff
@@ -974,9 +974,15 @@ func (c *Client) GetRawNotaryPool() (*result.RawNotaryPool, error) {
 }
 
 // GetBlockNotifications returns notifications from a block organized by trigger type.
-func (c *Client) GetBlockNotifications(blockHash util.Uint256, filters ...*neorpc.NotificationFilter) (*result.BlockNotifications, error) {
-	var resp = &result.BlockNotifications{}
-	if err := c.performRequest("getblocknotifications", []any{blockHash.StringLE(), filters}, resp); err != nil {
+func (c *Client) GetBlockNotifications(blockHash util.Uint256, filter *neorpc.NotificationFilter) (*result.BlockNotifications, error) {
+	var (
+		resp = &result.BlockNotifications{}
+		flt  = neorpc.NotificationFilter{}
+	)
+	if filter != nil {
+		flt = *filter.Copy()
+	}
+	if err := c.performRequest("getblocknotifications", []any{blockHash.StringLE(), flt}, resp); err != nil {
```

Comment:
This way you're going to have empty notification filter passed to RPC server in case of nil `filter`. And what we need in this case is just a blockhash. See more carefully how `ReceiveExecutionNotifications` works and do the same for this handler.

Reply:
Add an end-to-end test for this particular case to `pkg/services/rpcsrv/client_test.go` to ensure that nil filter will be handled properly by server.

Reply:
Not fixed.
```
params := []any{blockHash.StringLE()}
if flt != nil {
	params = append(params, *flt)
}
if err := c.performRequest("getblocknotifications", params, resp); err != nil {
...
```
And after a bit of thinking, I see that filter copy here may be omitted because filter won't likely be reused by the caller's code.